src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -675,8 +678,10 @@ CHIP_ERROR TimeSynchronizationServer::SetDSTOffset(const DataModel::DecodableLis
         return ClearDSTOffset();
     }
 
-    auto newDstL = dstL.begin();
-    size_t i     = 0;
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    uint64_t lastValidUntil = 0;
```

Comment:
The use of "goto exit" is pretty unfortunate.  It might have been better to just have two functions: one which returns on failures, and one which calls the other and rolls back the state if failure is returned...  But OK.